connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -220,6 +221,7 @@ public boolean startConnector(
         final WorkerConnector workerConnector;
         ClassLoader savedLoader = plugins.currentThreadLoader();
         try {
+            LoggingContext.forConnector(connName);
```

Comment:
How did we decide what to change logging context for vs not? For example, `connectorTaskConfigs` later in this file doesn't do this, but it will make a call into the connector which could potentially do logging without providing context as to which *instance* is logging. Similarly, `stopConnector` doesn't seem to switch logging context. (in particular, I found those by checking for where we're doing plugin swapping, which seems a likely indicator we want to change contexts)

Reply:
Good catch. I've tried to not change the herder, which is where some of the plugin swapping is done, but `connectorTaskConfigs` and `stopConnector` and `stopTask` all should have switched logging contexts.

I also adjusted where the logging context is created in some of the other methods, to be more inclusive of the whole Worker method, rather than reusing an existing try-block to minimize changes.

Reply:
@ewencp, ah, I remember why I moved away from AutoCloseable: In a try-with-resources statement, any `catch` or `finally` block is run _after_ the resources declared have been closed.

I still think it's cleaner, but it may require an outer try-with-resources block outside of an existing try-catch-finally block to ensure that all log messages within the existing catch and finally blocks are logged in the same context.